src/client/Microsoft.Identity.Client/AppConfig/ConfidentialClientApplicationBuilder.cs

```diff
@@ -97,6 +97,55 @@ public ConfidentialClientApplicationBuilder WithCertificate(X509Certificate2 cer
             return this;
         }
 
+
+        /// <summary>
+        /// Sets the certificate associated with the application
+        /// </summary>
+        /// <param name="certificate">The X509 certificate used as credentials to prove the identity of the application to Azure AD.</param>
+        /// <param name="withSendX5C"><c>true</c> if the x5c should be sent. Otherwise <c>false</c>.
+        /// The default is <c>false</c></param>
+        /// <remarks>You should use certificates with a private key size of at least 2048 bytes. Future versions of this library might reject certificates with smaller keys. </remarks>
+        public ConfidentialClientApplicationBuilder WithCertificate(X509Certificate2 certificate, bool withSendX5C)
+        {
+            if (certificate == null)
+            {
+                throw new ArgumentNullException(nameof(certificate));
+            }
+
+            if (!certificate.HasPrivateKey)
+            {
+                throw new MsalClientException(MsalError.CertWithoutPrivateKey, MsalErrorMessage.CertMustHavePrivateKey(nameof(certificate)));
+            }
+
+            Config.ClientCredentialCertificate = certificate;
+            Config.ConfidentialClientCredentialCount++;
+            return WithSendX5C(withSendX5C);
+        }
+
+        /// <summary>
+        /// Specifies if the x5c claim (public key of the certificate) should be sent to the STS.
+        /// Sending the x5c enables application developers to achieve easy certificate roll-over in Azure AD:
+        /// this method will send the public certificate to Azure AD along with all STS requests,
+        /// so that Azure AD can use it to validate the subject name based on a trusted issuer policy.
+        /// This saves the application admin from the need to explicitly manage the certificate rollover
+        /// (either via portal or PowerShell/CLI operation). For details see https://aka.ms/msal-net-sni
+        /// </summary>
+        /// <param name="withSendX5C"><c>true</c> if the x5c should be sent. Otherwise <c>false</c>.
+        /// The default is <c>false</c></param>
+        /// <returns>The builder to chain the .With methods</returns>
+        public ConfidentialClientApplicationBuilder WithSendX5C(bool withSendX5C)
+        {
+            // Default is not to send X5C, so if called with false, this is a no-op. 
+            if (!withSendX5C)
+            {
+                return this;
+            }
+
+            Config.SendX5C= withSendX5C;
+            Config.ConfidentialClientCredentialCount++;
```

Comment:
Does this work? WithSendX5C should not increment this counter, which simply counts how many times APIs such as `WithSecret` `WithCertificate` etc. were used. It's used simply to throw an exception when 0 or >1 secrets are configured,.

Reply:
My thinking was that this would cause Build to throw if you had set WithSendX5C and not set a certificate. That would be expected, because you can only use x5c with a cert (and not with a secret or provided assertion).

Reply:
Yeah, I agree that if you set sendX5C but don't use WithCertificate, we should throw. I don't think we do this now, and there is no need to do it as part of this PR.

But when we throw exceptions, we need to be very explicit in describing the error message. In this case what would happen is that if the developer adds 

`app.WithCertificate(cert, true)` then that counter would be 2 and MSAL would throw an exception thinking "oh, you've used WithCertificate and WithSecret at the same time, this is not supported".

Reply:
Oh, I totally misread this to be a tracker on cert use, not credentials attached. Yep, will remove. Thanks!